test/e2e_node/container_lifecycle_pod_construction.go

```diff
@@ -122,23 +122,30 @@ func (o containerOutputList) String() string {
 // RunTogether returns an error the lhs and rhs run together
 func (o containerOutputList) RunTogether(lhs, rhs string) error {
 	lhsStart := o.findIndex(lhs, "Started", 0)
-	rhsStart := o.findIndex(rhs, "Started", 0)
-
-	lhsFinish := o.findIndex(lhs, "Exiting", 0)
-	rhsFinish := o.findIndex(rhs, "Exiting", 0)
-
 	if lhsStart == -1 {
 		return fmt.Errorf("couldn't find that %s ever started, got\n%v", lhs, o)
 	}
+
+	rhsStart := o.findIndex(rhs, "Started", lhsStart+1)
```

Comment:
This will exclude the case that rhs starts before the lhs but they run together.

Could you explain what is the reason for changing this?

Reply:
this is more reproducible IMHO and if rhs starts first, just swap rhs and lhs (as it's done in the other file)

Reply:
I am little bit worried about the ordering part as we cannot use this in cases that we cannot guarantee the ordering (e.g. regular containers run together part?).

How about add another test function like RunTogetherWithOrdering instead?

Reply:
well today we don't use that with regular containers... so we could add another one RunTogetherWithoutOrdering?